src/wallet/asyncrpcoperation_sendmany.cpp

```diff
@@ -0,0 +1,972 @@
+        
+        // If there is a single zaddr and there are coinbase utxos, change goes to the zaddr.
+        if (change > 0) {
+            if (isSingleZaddrOutput && selectedUTXOCoinbase) {
+                std::string address = std::get<0>(zOutputsDeque.front());   
+                SendManyRecipient smr(address, change, std::string());
+                zOutputsDeque.push_back(smr);
+
+                LogPrint("asyncrpc", "%s: change from coinbase utxo is also sent to the recipient (amount=%s)\n",
+                        getId().substr(0, 10),
+                        FormatMoney(change, false)
+                        );
+
+            } else if (!isSingleZaddrOutput && selectedUTXOCoinbase) {
+                // This should not happen and is not allowed
+                throw JSONRPCError(RPC_WALLET_ERROR, "Wallet selected Coinbase UTXOs as valid inputs when it should not have done");
+            } else {
+                // If there is a single zaddr and no coinbase utxos, just use a regular output for change.
+                add_taddr_change_output_to_tx(change);
+
+                LogPrint("asyncrpc", "%s: transparent change in transaction output (amount=%s)\n",
+                        getId().substr(0, 10),
+                        FormatMoney(change, false)
+                        );
+            }
+        }
+
+        // Create joinsplits, where each output represents a zaddr recipient.
+        Object obj;
+        while (zOutputsDeque.size() > 0) {
+            AsyncJoinSplitInfo info;
+            info.vpub_old = 0;
+            info.vpub_new = 0;
+            int n = 0;
+            while (n++<2 && zOutputsDeque.size() > 0) {
+                SendManyRecipient smr = zOutputsDeque.front();
+                std::string address = std::get<0>(smr);
+                CAmount value = std::get<1>(smr);
+                std::string hexMemo = std::get<2>(smr);
+                zOutputsDeque.pop_front();
+
+                PaymentAddress pa = CZCPaymentAddress(address).Get();
+                JSOutput jso = JSOutput(pa, value);
+                if (hexMemo.size() > 0) {
+                    jso.memo = get_memo_from_hex_string(hexMemo);
+                }
+                info.vjsout.push_back(jso);
+                
+                // Funds are removed from the value pool and enter the private pool
+                info.vpub_old += value;
+            }
+            obj = perform_joinsplit(info);
+        }
+        sign_send_raw_transaction(obj);
+        return true;
+    }
+    /**
+     * END SCENARIO #2
+     */   
+ 
+    
+    
+    /**
+     * SCENARIO #3
+     * 
+     * zaddr -> taddrs
+     *       -> zaddrs
+     * 
+     * Processing order:
+     * Part 1: taddrs and miners fee
+     * Part 2: zaddrs 
+     */
+    
+    /**
+     * SCENARIO #3
+     * Part 1: Add to the transparent value pool.
+     */
+    Object obj;
+    CAmount jsChange = 0;   // this is updated after each joinsplit
```

Comment:
1. jsChange flows from Scenario 3 Part 1 to Part 2.  It's suboptimal right now as Part 1 should try to use one output for a zaddr recipient if there is one.
2. obj flows to the bottom of Scenario Part 2 where it is signed/sent.
I agree we should look at scope so variables don't scroll off the screen... but it's too close to release to make that change now.